social_media/accounts/views.py

```diff
@@ -0,0 +1,104 @@
+from django.contrib.auth import authenticate
+from django.shortcuts import render
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.authtoken.models import Token
+from .serializer import AccountRegistrationSerializer, AccountSerializer
+from .models import User
+
+
+# Create your views here.
+class UserDetailsView(APIView):
+
+    def get(self, request, username):
+        if User.objects.filter(username=username).exists():
+            user = User.objects.get(username=username)
+            account_serializer = AccountSerializer(instance=user)
+            return Response(account_serializer.data)
+        else:
+            return Response(data={'error': {'message': 'user not found'}}, status=status.HTTP_404_NOT_FOUND)
+
+
+class LoginView(APIView):
+    def post(self, request):
+        data = request.data
+        email = data.get('email', None)
+        password = data.get("password", None)
+        if not email or not password:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        # check if a user with such username or email exist
+        user = None
+        if User.objects.filter(email=email).exists():
+            user = User.objects.get(email=email)
+
+        else:
+            return Response({'error': 'Such user was not found'},
+                            status=status.HTTP_404_NOT_FOUND)
+        username = user.username
+        user = authenticate(username=username, password=password)
+        if not user:
```

Comment:
If authenticate will return None for incorrect username and password why not use it directly with the input from the request instead of checking first if the user exists or not

Reply:
@evilcudmovement Like Sir if user email does not exits we can send message like you dont have account you can create new one with this email